plugins/outputs/iotdb/iotdb_test.go

```diff
@@ -0,0 +1,377 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package iotdb
+
+import (
+	"fmt"
+	"math"
+	"testing"
+	"time"
+
+	"github.com/apache/iotdb-client-go/client"
+	"github.com/stretchr/testify/require"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/metric"
+	"github.com/influxdata/telegraf/testutil"
+)
+
+var (
+	testHost     = "localhost" // The server's (ip) address that you want to connect to.
+	testPort     = "6667"      // The server's port that you want to connect to.
+	testUser     = "root"
+	testPassword = "root"
+)
+
+func newTestClient() *IoTDB {
+	testClient := newIoTDB()
+	testClient.Host = testHost
+	testClient.Port = testPort
+	testClient.User = testUser
+	testClient.Password = testPassword
+	testClient.Log = testutil.Logger{}
+	return testClient
+}
+
+func TestConnectAndClose(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	testClient := newTestClient()
+
+	var err error
+	err = testClient.Connect()
+	require.NoError(t, err)
+	err = testClient.Close()
+	require.NoError(t, err)
+}
+
+func TestInitAndConnect(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test in short mode")
+	}
+	testClient := newTestClient()
+
+	var err error
+	err = testClient.Init()
+	require.NoError(t, err)
+	err = testClient.Connect()
+	require.NoError(t, err)
+	err = testClient.Close()
+	require.NoError(t, err)
+}
+
+func generateTestMetric(
+	name string,
+	tags []telegraf.Tag,
+	fields []telegraf.Field,
+	timestamp time.Time,
+) telegraf.Metric {
+	m := metric.New(name, map[string]string{}, map[string]interface{}{}, timestamp)
+	for _, tag := range tags {
+		m.AddTag(tag.Key, tag.Value)
+	}
+	for _, field := range fields {
+		m.AddField(field.Key, field.Value)
+	}
+	return m
+}
+
+var (
+	constTestTimestamp = time.Date(2022, time.July, 20, 12, 25, 33, 44, time.UTC)
+	testMetrics001     = []telegraf.Metric{
+		generateTestMetric(
+			"root.computer.fan",
+			[]telegraf.Tag{
+				{Key: "price", Value: "expensive"},
+				{Key: "owner", Value: "cpu"},
+			},
+			[]telegraf.Field{
+				{Key: "temperature", Value: float64(42.55)},
+				{Key: "counter", Value: int64(987654321)},
+			},
+			constTestTimestamp,
+		),
```

Comment:
Please use what we already have
```suggestion
		metric.New(
			"root.computer.fan",
			map[string]string{
				"price": "expensive",
				"owner": "cpu",
			},
			map[string]interface{}{
				"temperature": float64(42.55),
				"counter": int64(987654321),
			},
			time.Date(2022, time.July, 20, 12, 25, 33, 44, time.UTC),
		),
```

Reply:
All right. The code I wrote is really redundant. Thanks for reminding.